app/models/facility.rb

```diff
@@ -86,13 +86,25 @@ def siret_is_hardcoded_valid(siret)
     end
   end
 
-  ##
+  ## insee_code / commune helpers
   #
-  def to_s
-    "#{company.name} (#{commune_name})"
+  def insee_code=(insee_code)
+    self.commune = Commune.find_or_initialize_by(insee_code: insee_code)
+    city_params = ApiAdresse::Query.city_with_code(insee_code)
+    self.readable_locality = "#{city_params['codesPostaux']&.first} #{city_params['nom']}"
+  end
+
+  def insee_code
+    commune&.insee_code
```

Comment:
on peut écrire `delegate :insee_code, to: : commune`

Reply:
:+1:, avec allow_nil: true parce que les nouvelles facility pour les formulaires de nouvelle analyse n’ont pas de commune. De toute façon, les jours de Commune sont comptés. (comme dirait Thiers)